[DOC-4606] Release notes for v22.2.0-beta.4 (#15369)
cockroachdb/docs: _includes/releases/v22.2/v22.2.0-beta.4.md

## v22.2.0-beta.4

Release Date: October 17, 2022

{% include releases/release-downloads-docker-image.md release=include.release %}

<h3 id="v22-2-0-beta-4-db-console-changes">DB Console changes</h3>

- Fixed a bug that prevented usage of profiling links on the Advanced Debug page. [#89310][#89310]

<h3 id="v22-2-0-beta-4-bug-fixes">Bug fixes</h3>

- Fixed a bug in `Concat` projection operators for arrays that could cause non-null values to be added to the array when one of the arguments was null. [#89347][#89347]
- `VIEWCLUSTERMETADATA` permissions are now required to view hot ranges, problem ranges, data distribution, stores report, range status, and Raft for all ranges.  [#89509][#89509]
- Updated `SHOW CLUSTER SETTING` sql command to require `VIEWCLUSTERSETTING` or `MODIFYCLUSTERSETTING` permissions. [#89509][#89509]
- Fixed a visual bug on stores report where an error shows an infinite spinner only. [#89509][#89509]

> - Fixes tables created by userfile storage that have invalid foreign key constraints. [#89370][#89370]

Reviewer: This doesn't look properly worded. Perhaps it's something like "Fixed a bug in which tables created using [`userfile`](use-userfile-for-bulk-operations.html) would have invalid foreign key constraints." But this is a guess.

```suggestion
- Fixes tables created by userfile storage that have invalid foreign key constraints. [#89370][#89370]
```

> - A `VOTER_DEMOTING_LEARNER` can acquire the lease in a joint configuration only when there is a `VOTOR_INCOMING` in the configuration and the `VOTER_DEMOTING_LEARNER` was the last leaseholder. This prevents a situation in which the system is unable to exit the joint configuration. [#89595][#89595]

Reviewer: This doesn't state that the bug was fixed. It should be phrased "Previously, <faulty behavior>. This is now fixed."

> - Fixed a bug, introduced in a 22.2.x alpha version, in which uncommitted privileges could be cached if a transaction is rolled back. For example:

Reviewer: Per some previous team discussions on this, we don't need a full example within release note bullets, so the SQL block can be cut. People can refer to linked PR for details.

```suggestion
- Fixed a bug, introduced in a v22.2.x alpha version, in which uncommitted privileges could be cached if a transaction was rolled back.
```

    ~~~ sql
    BEGIN;
    GRANT SELECT ON crdb_internal.tables TO testuser;
    SELECT has_table_privilege('testuser', 'crdb_internal.tables', 'SELECT');
    --- this caches the privilege ---
    ROLLBACK;
    --- SELECT IS STILL CACHED UNTIL ANOTHER GRANT/REVOKE HAPPENS TO INVALIDATE THE CACHE---
    ~~~

    [#89718][#89718]

> - Fixed a bug introduced in a v22.2.0 beta version, that caused queries with expressions like `'foo' LIKE col` to return incorrect values, but only when an inverted trigram index exists on `col`. [#89701][#89701]

Reviewer: ```suggestion
- Fixed a bug introduced in a v22.2.0 beta version that caused queries with expressions like `'foo' LIKE col` to return incorrect values, but only when an inverted trigram index existed on `col`. [#89701][#89701]
```
